samwise.js

```diff
@@ -4,30 +4,28 @@
  * Copyright 2016 Henrik Ingo, henrik.ingo@avoinelama.fi
```

Comment:
SCRIPT-This function seems to have grown quite complex. Can we break it down further?